src/responder/nss/nss_protocol_grent.c

```diff
@@ -0,0 +1,424 @@
+
+    if (name == NULL || gid == 0) {
+        DEBUG(SSSDBG_OP_FAILURE, "Incomplete group object for %s[%u]! "
+              "Skipping\n", name ? name:"<NULL>", gid);
+        return EINVAL;
+    }
+
+    /* Convert to sized strings. */
+    ret = sized_output_name(mem_ctx, nss_ctx->rctx, name, domain, _name);
+    if (ret != EOK) {
+        DEBUG(SSSDBG_CRIT_FAILURE, "sized_output_name failed, skipping "
+              "[%d]: %s\n", ret, sss_strerror(ret));
+        return ret;
+    }
+
+    *_gid = gid;
+
+    return EOK;
+}
+
+static struct ldb_message_element *
+nss_get_group_members(struct sss_domain_info *domain,
+                      struct ldb_message *msg)
+{
+    struct ldb_message_element *el;
+
+    if (domain->ignore_group_members) {
+        return NULL;
+    }
+
+    /* Unconditionally prefer OVERRIDE_PREFIX SYSDB_MEMBERUID, it
+     * might contain override names from the default view. */
+    el = ldb_msg_find_element(msg, OVERRIDE_PREFIX SYSDB_MEMBERUID);
+    if (el == NULL) {
+        el = ldb_msg_find_element(msg, SYSDB_MEMBERUID);
+    }
+
+    return el;
+}
+
+static struct ldb_message_element *
+nss_get_group_ghosts(struct sss_domain_info *domain,
+                     struct ldb_message *msg,
+                     const char *group_name)
+{
+    struct ldb_message_element *el;
+
+    if (domain->ignore_group_members) {
+        return NULL;
+    }
+
+    el = ldb_msg_find_element(msg, SYSDB_GHOST);
+    if (el == NULL) {
+        return NULL;
+    }
+
+    if (DOM_HAS_VIEWS(domain) && !is_local_view(domain->view_name)
+            && el->num_values != 0) {
+        DEBUG(SSSDBG_CRIT_FAILURE,
+              "Domain has a view [%s] but group [%s] still has " \
+              "ghost members.\n", domain->view_name, group_name);
+            return NULL;
+    }
+
+    return el;
+}
+
+static errno_t
+nss_protocol_fill_members(struct sss_packet *packet,
+                          struct nss_ctx *nss_ctx,
+                          struct sss_domain_info *domain,
+                          struct ldb_message *msg,
+                          const char *group_name,
+                          size_t *_rp,
+                          uint32_t *_num_members)
+{
+    TALLOC_CTX *tmp_ctx;
+    struct resp_ctx *rctx = nss_ctx->rctx;
+    struct ldb_message_element *members[2];
+    struct ldb_message_element *el;
+    struct sized_string *name;
+    const char *member_name;
+    uint32_t num_members;
+    size_t body_len;
+    uint8_t *body;
+    errno_t ret;
+    int i, j;
+
+    tmp_ctx = talloc_new(NULL);
+    if (tmp_ctx == NULL) {
+        DEBUG(SSSDBG_FATAL_FAILURE, "Out of memory!\n");
+        return ENOMEM;
+    }
+
+    members[0] = nss_get_group_members(domain, msg);
+    members[1] = nss_get_group_ghosts(domain, msg, group_name);
+
+    sss_packet_get_body(packet, &body, &body_len);
+
+    num_members = 0;
+    for (i = 0; i < sizeof(members)/sizeof(members[0]); i++) {
+        el = members[i];
```

Comment:
it might be good have spaces around division
and from performance POV, you might store it to variable/constant
We needn't divide it every time.